runtime/util/hshelp.c

```diff
@@ -1701,6 +1701,72 @@ fixJNIRefs(J9VMThread * currentThread, J9HashTable * classPairs, BOOLEAN fastHCR
 }
 
 
+void
+fixMemberNames(J9JavaVM * vm, J9HashTable * classHashTable)
+{
+	PORT_ACCESS_FROM_JAVAVM(vm);
+
+	/* iterate over all objects fixing up vmtarget refs if object is a MemberName */
+	vm->memoryManagerFunctions->j9mm_iterate_all_objects(vm, PORTLIB, 0, fixMemberNamesObjectIteratorCallback, classHashTable);
+}
+
+static jvmtiIterationControl
+fixMemberNamesObjectIteratorCallback(J9JavaVM *vm, J9MM_IterateObjectDescriptor *objectDesc, void *userData)
+{
+	J9HashTable *classHashTable = userData;
+	j9object_t object = objectDesc->object;
+	J9Class *clazz = J9OBJECT_CLAZZ_VM(vm, object);
+
+	if (NULL != classHashTable) {
+		if (clazz == J9VMJAVALANGINVOKEMEMBERNAME(vm) && NULL != J9OBJECT_U64_LOAD(vm->currentThread, object, vm->vmindexOffset)) {
+			j9object_t vmindex = J9OBJECT_U64_LOAD(vm->currentThread, object, vm->vmindexOffset);
+			J9Class *indexClazz = J9OBJECT_CLAZZ_VM(vm, vmindex);
+
+			J9JVMTIClassPair exemplar;
+			J9JVMTIClassPair *result = NULL;
+
+			exemplar.originalRAMClass = J9VMJAVALANGINVOKEMEMBERNAME_CLAZZ(vm->currentThread, object);
+			result = hashTableFind(classHashTable, &exemplar);
+			if (NULL != result) {
+				J9Class *replacementRAMClass = result->replacementClass.ramClass;
+				if (NULL != replacementRAMClass) {
+					if (indexClazz == J9VMJAVALANGREFLECTFIELD(vm)) {
```

Comment:
use the `MemberName.flags` to check whether the MN is a field or method/constructor (flags are defined in [MethodHandleNatives.cpp](https://github.com/eclipse-openj9/openj9/blob/master/runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp#L47))

see here for usage: https://github.com/eclipse-openj9/openj9/blob/master/runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp#L568

Reply:
Won't I have to #include that cpp file? MN_IS_FIELD etc are not visible from hshelp.c